src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -542,11 +526,12 @@ private CustomResourceSnapshot ToSnapshot(Container container, CustomResourceSna
         var urls = GetUrls(container);
 
         var environment = GetEnvironmentVariables(container.Status?.EffectiveEnv ?? container.Spec.Env, container.Spec.Env);
+        var state = container.AppModelInitialState is "Hidden" ? "Hidden" : container.Status?.State;
```

Comment:
I'm not a fan of magic strings throughout logic.

Can `Hidden` be made a known state? Do were have a static type with known states? Or add an `IsHidden` method.

Reply:
We don’t have one but I agreee we should have one for known states. The public API in this area is a bit sketchy as well

Reply:
Done